File: 프로그래머스/unrated/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    result = ""
    #성격 유형 딕셔너리 화
    personal = {"RT":0,"CF":0,"JM":0,"AN":0}
    #성격유형점수 산정을 위한 연산
    for i in range(len(choices)):
        choices[i] -= 4
    #성격유형 점수 연산
    for i in range(len(survey)):
        if survey[i] in personal:
            personal[survey[i]] += choices[i]
        else:
            personal[survey[i][::-1]] -= choices[i]
    #성격유형 판별
    for k, v in personal.items():
        if v > 0:
            result += k[1]
        else:
            result += k[0]
    return result

File: 프로그래머스/unrated/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        //성격 유형 딕셔너리 화
        HashMap<String, Integer> hashSurvey = new HashMap<>();
        hashSurvey.put("RT",0);
        hashSurvey.put("CF",0);
        hashSurvey.put("JM",0);
        hashSurvey.put("AN",0);
        //성격유형점수 산정을 위한 연산
        for (int i = 0; i < choices.length; i++) choices[i] -= 4;
        //성격유형점수 연산
        for (int i = 0; i < survey.length; i++){
            if (hashSurvey.containsKey(survey[i])) hashSurvey.put(survey[i],hashSurvey.get(survey[i])+choices[i]);
            else{
                StringBuffer sb = new StringBuffer(survey[i]);
                String s_rev = sb.reverse().toString();
                hashSurvey.put(s_rev,hashSurvey.get(s_rev)-choices[i]);
            }
        }
        //성격유형 판별
        for (String ss : hashSurvey.keySet()){
            if (hashSurvey.get(ss) > 0) answer += ss.charAt(1);
            else answer += ss.charAt(0); 
        }
        return answer;
    }
}